networkx/readwrite/mermaid.py

```diff
@@ -0,0 +1,238 @@
+
+    for u, v in G.edges(data=False):
+        _handle_ids(u, ids_mapping, ids_generator)
+        _handle_ids(v, ids_mapping, ids_generator)
+        final_u = _format_node_id(u, ids_mapping)
+        final_v = _format_node_id(v, ids_mapping)
+        yield f"{final_u} --> {final_v}"
+
+
+@open_file(1, mode="wb")
+def write_mermaid(G, path, encoding="utf-8"):
+    """Write graph as a mermaid flowchart.
+
+    Parameters
+    ----------
+    G : graph
+       A NetworkX graph
+    path : file or string
+       File or filename to write. If a file is provided, it must be
+       opened in 'wb' mode. Filenames ending in .gz or .bz2 will be compressed.
+    encoding: string, optional
+       Specify which encoding to use when writing file.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(4)
+    >>> nx.write_mermaid(G, "test.mermaid")
+    >>> G = nx.path_graph(4)
+    >>> fh = open("test.mermaid", "wb")
+    >>> nx.write_mermaid(G, fh)
+    >>> nx.write_mermaid(G, "test.mermaid.gz")
+
+    See Also
+    --------
+    read_mermaid
+    """
+    path.write("flowchart\n".encode(encoding))
+    for line in generate_mermaid(G):
+        line = f"    {line}\n"
+        path.write(line.encode(encoding))
+
+
+@nx._dispatchable(graphs=None, returns_graph=True)
+def parse_mermaid(lines):
+    """Parse lines of a mermaid flowchart representation of a graph.
+
+    Parameters
+    ----------
+    lines : list or iterator of strings
+        Input data in mermaid flowchart format
+
+    Returns
+    -------
+    G: NetworkX Graph
+        The graph corresponding to lines
+
+    Examples
+    --------
+    Mermaid flowchart:
+
+    >>> lines = ["flowchart", "A --> B", "A --> C", "B --> C"]
+    >>> G = nx.parse_mermaid(lines)
+    >>> list(G)
+    ['A', 'B', 'C']
+    >>> list(G.edges())
+    [('A', 'B'), ('A', 'C'), ('B', 'C')]
+    """
+    G = nx.empty_graph(0)
+    skip_lines = True
+    for line in lines:
+        if "flowchart" in line:
+            skip_lines = False
+            continue
+        if skip_lines:
+            continue
+        if "-->" in line:
+            parts = line.strip().split(" ")
+            u = parts[0]
+            v = parts[-1]
+            G.add_edge(u, v)
+    return G
+
+
+@open_file(0, mode="rb")
+@nx._dispatchable(graphs=None, returns_graph=True)
+def read_mermaid(path, encoding="utf-8"):
+    """Read a graph from a list of edges.
+
+    Parameters
+    ----------
+    path : file or string
+       File or filename to read. If a file is provided, it must be
+       opened in 'rb' mode.
+       Filenames ending in .gz or .bz2 will be uncompressed.
+    encoding: string, optional
+       Specify which encoding to use when reading file.
+
+    Returns
+    -------
+    G : graph
+       A networkx Graph
+
+    Examples
+    --------
+    >>> nx.write_mermaid(nx.path_graph(4), "test.mermaid")
+    >>> G = nx.read_mermaid("test.mermaid")
+    >>> fh = open("test.mermaid", "rb")
```

Comment:
This suggestion adds back a blank line that mistakenly removed.
```suggestion

    >>> fh = open("test.mermaid", "rb")
```